File: SpO2/MovingAverageFilter.py
import numpy as np


def MovingAverageFilter(inputPPG, windowSize):
    window = np.zeros(windowSize)
    size_in = len(inputPPG)
    p_data = 0
    output = np.zeros(size_in)

    for i in range(size_in):
        window[p_data] = inputPPG[i]
        output[i] = np.sum(window) / windowSize
        p_data += 1
        if p_data >= windowSize:
            p_data = 0

    output[:windowSize] = inputPPG[:windowSize]
    # Keep the output constant when the number of input data is less than windowSize
    return output

File: SpO2/MATLAB/MovingAverageFilter.m
%moving average
function [output] = MovingAverageFilter(inputPPG,windowSize)

window = zeros(1,windowSize);
size_in = size(inputPPG);
p_data = 1;
output = zeros(size_in);
 
for i = 1:max(size_in)
    window(p_data) = inputPPG(i);
    output(i) = sum(window)/windowSize;
    p_data = p_data+1;
    if p_data > windowSize
        p_data = 1;
    end
end

output(1:windowSize) = inputPPG(1:windowSize);%Keep the output constant when the number of input data is less than windowSize
 
end
